### PR TITLE
Add Annotation layer parameter type to processing, new algorithm for converting main annotation layer items to secondary annotation layer

### PR DESCRIPTION
**Pull request by nyalldawson**
This is useful for moving items created in the main layer to a secondary layer, so that the item placement can be adjusted within the
layer stack

### CODE REVIEW

File: src/analysis/processing/qgsnativealgorithms.cpp [hunk 1]
```diff
@@ -477,6 +478,7 @@ void QgsNativeAlgorithms::loadAlgorithms()
   addAlgorithm( new QgsTaperedBufferAlgorithm() );
   addAlgorithm( new QgsTinMeshCreationAlgorithm() );
   addAlgorithm( new QgsTransectAlgorithm() );
+  addAlgorithm( new QgsTransferAnnotationsFromMainAnnotationLayerAlgorithm() );
```
Comment (nirvn) [inline src/analysis/processing/qgsnativealgorithms.cpp:481]: @nyalldawson , I think that class name is a bit too short... :wink:

File: src/core/processing/qgsprocessingalgorithm.h [hunk 2]
```diff
@@ -988,6 +988,16 @@ class CORE_EXPORT QgsProcessingAlgorithm
      */
     QgsPointCloudLayer *parameterAsPointCloudLayer( const QVariantMap &parameters, const QString &name, QgsProcessingContext &context ) const;
 
+    /**
+     * Evaluates the parameter with matching \a name to an annotation layer.
+     *
+     * Annotation layers will either be taken from \a context's active project. Callers do not
```
Comment (nirvn) [inline src/core/processing/qgsprocessingalgorithm.h:994]:
```suggestion
     * Annotation layers will be taken from \a context's active project. Callers do not
```

File: src/core/processing/qgsprocessingparameters.h [hunk 3]
```diff
@@ -1593,6 +1596,25 @@ class CORE_EXPORT QgsProcessingParameters
      */
     static QgsPointCloudLayer *parameterAsPointCloudLayer( const QgsProcessingParameterDefinition *definition, const QVariant &value, QgsProcessingContext &context );
 
+    /**
+     * Evaluates the parameter with matching \a definition to an annotation layer.
+     *
+     * Layers will either be taken from \a context's active project. Callers do not
```
Comment (nirvn) [inline src/core/processing/qgsprocessingparameters.h:1602]:
```suggestion
     * Layers will be taken from \a context's active project. Callers do not
```

File: src/core/processing/qgsprocessingparameters.h [hunk 4]
```diff
@@ -1593,6 +1596,25 @@ class CORE_EXPORT QgsProcessingParameters
...
+     * need to handle deletion of the returned layer.
+     *
+     * \since QGIS 3.22
+     */
+    static QgsAnnotationLayer *parameterAsAnnotationLayer( const QgsProcessingParameterDefinition *definition, const QVariantMap &parameters, QgsProcessingContext &context );
+
+    /**
+     * Evaluates the parameter with matching \a definition and \a value to an annotation layer.
+     *
+     * Layers will either be taken from \a context's active project. Callers do not
```
Comment (nirvn) [inline src/core/processing/qgsprocessingparameters.h:1612]:
```suggestion
     * Layers will be taken from \a context's active project. Callers do not
```

File: tests/src/analysis/testqgsprocessing.cpp [hunk 5]
```diff
@@ -998,20 +1018,20 @@ void TestQgsProcessing::compatibleLayers()
...
 #endif
 
   // unsorted
   lIds.clear();
   for ( QgsMapLayer *l : QgsProcessingUtils::compatibleLayers( &p, false ) )
     lIds << l->name();
-#ifdef WITH_EPT
-  QCOMPARE( lIds, QStringList() << "R1" << "ar2" << "zz"  << "V4" << "v1" << "v3" << "vvvv4" << "MX" << "mA" << "PCX" << "pcA" << "PX" << "pA" );
+#ifdef HAVE_EPT
+  QCOMPARE( lIds, QStringList() << "R1" << "ar2" << "zz"  << "V4" << "v1" << "v3" << "vvvv4" << "MX" << "mA" << "PCX" << "pcA" << "secondary annotation layer" << "Annotations" << "PX" << "pA" );
```
Comment (nirvn) [inline tests/src/analysis/testqgsprocessing.cpp:1032]: Is "Annotations" going to fail when someone runs this test on a non-English locale? I.e., should it be QObject::tr( "Annotations" ) ?

Comment (nyalldawson) [reply tests/src/analysis/testqgsprocessing.cpp:1032]: I've done that -- but if the translation changes the sort order then the test will still break unfortunately

**Pull request merged**